Tweak a few file to allow WordPressData to compile

mokagio opened the pull request:
Discovered these while working on #24378.

Part of #24165.

mokagio committed:
Move code `localIconImage` to app targets / Keystone

This way, when `PublicizeService` moves to WordPressData, it won't need
the dependency used in that code.

mokagio committed:
Use `BuildSettings` over `AppConfiguration` in `SharedDataIssueSolver`

mokagio committed:
Use `BuildSettings` over `AppConfiguration` in `WPAccount+Lookup`

mokagio committed:
Use `_private_wordPressComRestApi` in `Blog.m` to work around deps

mokagio committed:
Properly call `alloc` in `PostService.m`

dangermattic commented:
<!--
  0 Errors
  1 Warning: This PR is assigned to the mil...
  0 Messages
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/wordpress-mobile/WordPress-iOS/milestone/308">25.9</a>. The due date for this milestone has already passed.<br />
Please assign it to a milestone with a later deadline or check whether the release for this milestone has already been finished.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

mokagio commented on WordPress/Classes/Jetpack/Utility/SharedDataIssueSolver.swift:
```diff
@@ -41,7 +41,7 @@ public final class SharedDataIssueSolver: NSObject {
     ///
     public func migrateAuthKey(for username: String) {
-        guard AppConfiguration.isJetpack,
+        guard BuildSettings.current.brand == .jetpack,
```
Using `BuildSettings` over `AppConfiguration` allows moving to WordPressData where the latter is not available.

mokagio commented on WordPress/Classes/Models/Blog/Blog.m:
```diff
@@ -908,7 +908,7 @@ - (WordPressOrgRestApi *)selfHostedSiteRestApi
 {
     if (self.account) {
-        return self.account.wordPressComRestApi;
+        return self.account._private_wordPressComRestApi;
```
`wordPressComRestApi` uses `WordPressAuthenticationManager` which depends on WordPressAuthenticator which we don't want to add to WordPressData.

kean replied:
Does this method need to be in `WordPressData`? Would it make sense to keep inside the app targets like the similar `WPAccount` extension?

mokagio replied:
Good question... I'll have to check.

mokagio replied:
After a quick check, I think it should be possible to move it to 1) Swift and 2) the apps targets. This will also require moving some additional logic for logging / Zendesk / customer support, but it should not be too complex...

mokagio replied:
Addressed via [`75f416c` (#24375)](https://github.com/wordpress-mobile/WordPress-iOS/pull/24375/commits/75f416c94ee32610cce5e6adf200108e27498cd7) @kean — thanks again for the suggestion

mokagio commented on WordPress/Classes/Models/PublicizeService.swift:
```diff
@@ -41,11 +41,6 @@ public extension PublicizeService {
         case threads
         case unknown
 
-        /// Returns the local image for the icon representing the social network.
-        var localIconImage: UIImage {
-            WPStyleGuide.socialIcon(for: rawValue as NSString)
-        }
-
```
`socialIcon(for service: NSString) -> UIImage` is defined in the apps target / Keystone, which WordPressData does not have access to. 

By moving it away from here, we unblock adding this required extension to the framework.

mokagio committed:
Move `Blog` `wordPressComRestApi` and consumers from ObjC to Swift

Apologies for the annoying diff due to the change from methods to
computed vars.

I suppose I could have left the implementations as method, but it felt
cleaner to move them to more Swift-idiomatic computed vars.

mokagio commented:
The tests [failed](https://buildkite.com/automattic/wordpress-ios/builds/26984) in CI before [`75f416c` (#24375)](https://github.com/wordpress-mobile/WordPress-iOS/pull/24375/commits/75f416c94ee32610cce5e6adf200108e27498cd7). For the record, for `75f416c` they pass on my end... but you'll have to take my word for it because my Xcode just quit itself to update...

<img width="372" alt="Screenshot 2025-04-01 at 11 03 22 PM" src="https://github.com/user-attachments/assets/8f9e4b83-ac58-4649-94a4-2a0651ed923d" />

kean reviewed (approved):
LGTM. If you need something rewriting from Objective-C, let me know.
